sdk/core/core-amqp/src/retry.ts

```diff
@@ -219,21 +219,23 @@ export async function retry<T>(config: RetryConfig<T>): Promise<T> {
         i,
         err
       );
-      let targetDelayInMs = config.retryOptions.retryDelayInMs;
-      if (config.retryOptions.mode === RetryMode.Exponential) {
-        let incrementDelta = Math.pow(2, i) - 1;
-        const boundedRandDelta =
-          config.retryOptions.retryDelayInMs * 0.8 +
-          Math.floor(
-            Math.random() *
-              (config.retryOptions.retryDelayInMs * 1.2 - config.retryOptions.retryDelayInMs * 0.8)
-          );
-        incrementDelta *= boundedRandDelta;
-
-        targetDelayInMs = Math.min(incrementDelta, config.retryOptions.maxRetryDelayInMs);
-      }
 
       if (lastError && lastError.retryable && totalNumberOfAttempts > i) {
+        let targetDelayInMs = config.retryOptions.retryDelayInMs;
+        if (config.retryOptions.mode === RetryMode.Exponential) {
+          let incrementDelta = Math.pow(2, i) - 1;
+          const boundedRandDelta =
+            config.retryOptions.retryDelayInMs * 0.8 +
+            Math.floor(
+              Math.random() *
+                (config.retryOptions.retryDelayInMs * 1.2 -
+                  config.retryOptions.retryDelayInMs * 0.8)
+            );
+          incrementDelta *= boundedRandDelta;
+
+          targetDelayInMs = Math.min(incrementDelta, config.retryOptions.maxRetryDelayInMs);
```

Comment:
In that case, shouldnt the next two statements of logging the delay and actually running the delay also be inside this if block?

Reply:
They are inside the `if (lastError && lastError.retryable && totalNumberOfAttempts > i)` block.  The nested `if` is overwriting `targetDelayInMs` with exponential delay.